main.go

```diff
@@ -70,11 +73,32 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func runPrometheusServer(metricsAddr string) {
+	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
+	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", handler)
+	server := http.Server{
+		Addr:    metricsAddr,
+		Handler: mux,
+	}
+
+	go func() {
+		err := server.ListenAndServeTLS(path.Join(sdkTLSDir, sdkTLSCrt), path.Join(sdkTLSDir, sdkTLSKey))
+		if err != nil {
+			setupLog.Error(err, "Failed to start Prometheus metrics endpoint server")
+		}
+	}()
+}
+
+// Give permissions to use leases for leader election.
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch;delete
```

Comment:
This `kubebuilder` annotation is no longer needed. It is probably an artifact of rebasing this PR. You can remove it.